Objects/typeobject.c

```diff
@@ -2377,6 +2377,15 @@ type_new(PyTypeObject *metatype, PyObject *args, PyObject *kwds)
         nbases = 1;
     }
     else {
+        for (i = 0; i < nbases; i++) {
+            tmp = PyTuple_GET_ITEM(bases, i);
+            if (!PyType_Check(tmp) && PyObject_GetAttrString(tmp, "__mro_entries__")) {
```

Comment:
Are you leaking a reference to the object returned by GetAttrString here?

Reply:
Indeed, thank you for finding this!